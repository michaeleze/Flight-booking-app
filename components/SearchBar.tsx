import React from 'react';

type SearchBarProps = {
  headerText?: string;
};

const SearchBar: React.FC<SearchBarProps> = ({
  headerText = 'Search Flights',
}) => {
  return (
    <div className="my-8 mx-auto flex max-w-5xl flex-col space-x-4 rounded-xl bg-white p-6 shadow-md">
      <h1 className="pb-16 text-3xl">{headerText}</h1>
      <div>
        <input
          className="bg-gray-20 my-6 max-w-2xl rounded border border-gray-100 border-gray-100 p-2 text-gray-900"
          type="text"
          placeholder="FROM"
        />
        <input
          className="bg-gray-20 my-6 max-w-2xl rounded border border-gray-100 border-gray-100 p-2 text-gray-900"
          type="text"
          placeholder="TO"
        />
        <div className="relative">
          <div className="pointer-events-none absolute inset-y-0 left-0 flex items-center pl-3">
            <svg
              className="h-5 w-5 text-gray-500 dark:text-gray-400"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                fill-rule="evenodd"
                d="M6 2a1 1 0 00-1 1v1H4a2 2 0 00-2 2v10a2 2 0 002 2h12a2 2 0 002-2V6a2 2 0 00-2-2h-1V3a1 1 0 10-2 0v1H7V3a1 1 0 00-1-1zm0 5a1 1 0 000 2h8a1 1 0 100-2H6z"
                clip-rule="evenodd"
              ></path>
            </svg>
          </div>
          <input
            type="date"
            className="bg-gray-20 block w-2/5 rounded-lg border border-gray-100 p-2.5 pl-10 text-gray-900 focus:border-blue-500 focus:ring-blue-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-blue-500 dark:focus:ring-blue-500 sm:text-sm"
            placeholder="Select date"
          />
        </div>
        <div className="relative">
          <div className="pointer-events-none absolute inset-y-0 left-0 flex items-center pl-3">
            <svg
              className="h-5 w-5 text-gray-500 dark:text-gray-400"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                fill-rule="evenodd"
                d="M6 2a1 1 0 00-1 1v1H4a2 2 0 00-2 2v10a2 2 0 002 2h12a2 2 0 002-2V6a2 2 0 00-2-2h-1V3a1 1 0 10-2 0v1H7V3a1 1 0 00-1-1zm0 5a1 1 0 000 2h8a1 1 0 100-2H6z"
                clip-rule="evenodd"
              ></path>
            </svg>
          </div>
          <input
            type="date"
            className="bg-gray-20 block w-2/5 rounded-lg border border-gray-100 p-2.5 pl-10 text-gray-900 focus:border-blue-500 focus:ring-blue-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-blue-500 dark:focus:ring-blue-500 sm:text-sm"
            placeholder="Select date"
          />
        </div>
      </div>
    </div>
  );
};

export default SearchBar;
