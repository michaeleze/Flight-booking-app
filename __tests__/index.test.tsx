import {
  act,
  cleanup,
  render,
  RenderResult,
  fireEvent,
} from '@testing-library/react';
import Home from '@/pages/index';
import SearchBar from '../components/SearchBar';

const handleSearch = jest.fn(
  (
    origin = 'FRA',
    destination = 'FCO',
    departureDate = '2020-10-01',
    returnDate = '2020-10-02',
    cabinCode = 'economy'
  ) => null
);

describe('Home', () => {
  let component: RenderResult<
    typeof import('@testing-library/dom/types/queries'),
    HTMLElement
  >;

  beforeEach(() => {
    act(() => {
      component = render(<Home />);
    });
  });

  afterEach(cleanup);

  it('should match snapshot', async () => {
    const HomeComponent = await component;
    expect(HomeComponent).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    // const { getByTestId } = render(<SearchBar handleSearch={handleSearch} />);
    const { getByText, getByTestId } = component;

    fireEvent.change(getByTestId('select-Departure'), {
      target: { value: 'FRA' },
    });
    fireEvent.change(getByTestId('select-Destination'), {
      target: { value: 'MEX' },
    });
    fireEvent.click(getByTestId('search-button'));

    expect(getByText('FRA')).toBeInTheDocument();
  });
});
