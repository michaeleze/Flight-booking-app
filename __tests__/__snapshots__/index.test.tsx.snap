// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container m-6 mx-auto"
        data-testid="home"
      >
        <div
          class="container group relative z-0 mx-auto flex max-w-5xl flex-col space-x-4 rounded-xl bg-white p-6 shadow-md"
        >
          <div
            class="flex gap-3 pb-10"
          >
            <svg
              fill="currentColor"
              height="28"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="28"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M21 16v-2l-8-5V3.5c0-.83-.67-1.5-1.5-1.5S10 2.67 10 3.5V9l-8 5v2l8-2.5V19l-2 1.5V22l3.5-1 3.5 1v-1.5L13 19v-5.5l8 2.5z"
              />
            </svg>
            <h1
              class="text-3xl font-bold"
            >
              Flight Booking
            </h1>
          </div>
          <div
            class="flex flex-col gap-12 py-5 md:flex-row"
          >
            <div
              class="container md:w-1/2"
            />
            <div
              class="container md:w-1/2"
            />
          </div>
          <div
            class="flex flex-col gap-12 py-5 md:flex-row"
          >
            <div
              class="group container relative z-0 mb-6 md:w-1/3"
            />
            <div
              class="group container relative z-0 mb-6 md:w-1/3"
            />
            <div
              class="group relative z-0 mb-6 md:w-1/3"
            />
          </div>
          <button
            class=" my-8 self-center rounded-full bg-blue-700 px-5 py-2.5 text-center text-sm font-medium text-white hover:bg-blue-800 focus:outline-none focus:ring-4 focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800 md:w-1/4 md:self-end"
            data-testid="search-button"
            type="button"
          >
            Search flights
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container m-6 mx-auto"
      data-testid="home"
    >
      <div
        class="container group relative z-0 mx-auto flex max-w-5xl flex-col space-x-4 rounded-xl bg-white p-6 shadow-md"
      >
        <div
          class="flex gap-3 pb-10"
        >
          <svg
            fill="currentColor"
            height="28"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M21 16v-2l-8-5V3.5c0-.83-.67-1.5-1.5-1.5S10 2.67 10 3.5V9l-8 5v2l8-2.5V19l-2 1.5V22l3.5-1 3.5 1v-1.5L13 19v-5.5l8 2.5z"
            />
          </svg>
          <h1
            class="text-3xl font-bold"
          >
            Flight Booking
          </h1>
        </div>
        <div
          class="flex flex-col gap-12 py-5 md:flex-row"
        >
          <div
            class="container md:w-1/2"
          />
          <div
            class="container md:w-1/2"
          />
        </div>
        <div
          class="flex flex-col gap-12 py-5 md:flex-row"
        >
          <div
            class="group container relative z-0 mb-6 md:w-1/3"
          />
          <div
            class="group container relative z-0 mb-6 md:w-1/3"
          />
          <div
            class="group relative z-0 mb-6 md:w-1/3"
          />
        </div>
        <button
          class=" my-8 self-center rounded-full bg-blue-700 px-5 py-2.5 text-center text-sm font-medium text-white hover:bg-blue-800 focus:outline-none focus:ring-4 focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800 md:w-1/4 md:self-end"
          data-testid="search-button"
          type="button"
        >
          Search flights
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
